.github/workflows/main.yml

```diff
@@ -0,0 +1,239 @@
+          python jenkins_scripts/registry.py push cpp_backends --username "$DOCKER_USERNAME" --registry "$DOCKER_REGISTRY"
+      - name: C++ backends release
+        if: github.ref == 'refs/heads/releases'
+        run: |
+          python jenkins_scripts/registry.py build cpp_backends --dir docker --username "$DOCKER_USERNAME" --branch "$GIT_BRANCH"  --registry "$DOCKER_REGISTRY"
+          python jenkins_scripts/registry.py push cpp_backends --username "$DOCKER_USERNAME" --registry "$DOCKER_REGISTRY"
+
+  release:
+    needs: publish_images
+    runs-on: ubuntu-latest
+    if: success()
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.DOCKER_REGISTRY }}
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Switch to Current Branch
+        run: git checkout ${{ env.GIT_BRANCH }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          cd jenkins_scripts
+          pip install -r requirements.txt
+
+      - name: Set environment variables
+        run: |
+          echo "GIT_SHA=${GITHUB_SHA}" >> $GITHUB_ENV
+          echo "GIT_BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "GITHUB_USER: ${{ github.repository_owner }}"
+          echo "$GIT_BRANCH"
+
+      - name: Release
+        run: |
+          # Fetch most recent tags
+          git fetch --prune --unshallow --tags
+          echo $GIT_BRANCH
+          BUILD_TAG=$(python jenkins_scripts/github_release.py get-tag -b $GIT_BRANCH)
+          cd nms_stack
+          ANSIBLE_VERSION=2.9.9
+          OUTPUT_FILE="nms.pex"
+          SYMLINK="nms"
+
+          # Construct version file
+          echo "The build tag is: ${BUILD_TAG}"
+          echo "${BUILD_TAG}" > nms_cli/__version__
+
+          # push versions to console for debugging
+          echo "Packages"
+          pip freeze
+          pex --version
+
+          # build nms dist
+          echo "build nms dist"
+          python setup.py bdist_wheel
+
+          # build pex
+          echo "Build pex"
+          pex --wheel --python=python3 -v --disable-cache -f dist -f . ansible==$ANSIBLE_VERSION -e nms_cli.main -o ${OUTPUT_FILE} .
+
+          # pex sanity test
+          # echo "pex sanity test"
+          # ls -lah ${OUTPUT_FILE}
+          # ./${OUTPUT_FILE} --version
+
+          NOW=$(date +"%m_%d_%y_%H_%M_%S")
+          DATED_FILE="/tmp/nms_${NOW}"
+          mv ${OUTPUT_FILE} ${DATED_FILE}
+          rm -f ${SYMLINK}
+          ln -s ${DATED_FILE} ${SYMLINK}
+          echo "Check if all required images have been created:"
+          if ./nms check-images --nonfatal-image-check
```

Comment:
should we make the image check fatal as the TODO comment says, or should we ignore this for now since things are so in-flux?

Reply:
I think we should ignore this for now